packages/kit/src/runtime/app/forms.js

```diff
@@ -161,14 +165,29 @@ export function enhance(form_element, submit = () => {}) {
 		let result;
 
 		try {
+			const is_valid_enctype = enctype?.match(
+				/application\/x-www-form-urlencoded|multipart\/form|text\/plain/
+			);
+
+			const headers = {
+				accept: 'application/json',
+				'Content-Type': is_valid_enctype ? enctype : 'application/x-www-form-urlencoded',
+				'x-sveltekit-action': 'true'
+			};
+
+			// @ts-expect-error
+			const body = enctype === 'multipart/form-data' ? form_data : new URLSearchParams(form_data);
+
+			if (enctype === 'multipart/form-data' && headers['Content-Type']) {
+				// @ts-expect-error
+				delete headers['Content-Type'];
```

Comment:
We could avoid the TypeScript error here (for accessing the array index with a string key) by either using the [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers/Headers) object for `headers` or by typing the object key as a string such as:
```js
/** @type {Record<string, string> */
const headers = { ... }
```

Reply:
i've updated it to use `Headers` constructor